src/containers/VideoContainer/VideoContainer.js

```diff
@@ -5,26 +5,66 @@ import { View } from 'react-native'
 import { Button, Text } from 'react-native-paper'
 import { Video } from 'expo-av'
 import PropTypes from 'prop-types'
+import { FILE_SERVER_BASE_URL } from '../../constants'
+import { STACK_SCREENS } from '../CTNavigationContainer/index'
 import styles from './VideoContainer.style'
 
 // CT requires this header as an addititonal security measure. Since we're not an approved referer, we can actually
 // just hardcode a valid referer and the API accepts it.
 const REFERER =
   'https://classtranscribe-dev.ncsa.illinois.edu/video?id=c79700ac-c3fc-439f-95c2-0511a1092862'
 
-const VideoContainer = ({ url }) => {
+const VideoContainer = ({ videos, index, navigation }) => {
+  // Passing a list of videos would not be a bad approach because react pass object as reference
+  const url = videos[index].video.video1Path
```

Comment:
Can you use `videos[index]?.video?.video1Path`. This will prevent us from crashing if video1Path doesn't exist.